### PR TITLE
RoiMiniBatch enhancement

### PR DESCRIPTION
**Pull request by Menooker**
## What changes were proposed in this pull request?

Use view instead of slice. Correctly handle the case when label/image size info is empty.

## How was this patch tested?

UT

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/transform/vision/image/MTImageFeatureToBatch.scala [hunk 1]
```diff
@@ -269,8 +270,8 @@ object RoiImageInfo {
  * RoiImageInfo.IMGINFO  with shape (batchSize, 4), contains all images info
  *                 (height, width, original height, original width)
  */
-class RoiMiniBatch(val input: Tensor[Float], val target: Array[RoiLabel],
-  val isCrowd: Array[Tensor[Float]], val imageInfo: Tensor[Float] = null)
+class RoiMiniBatch(val input: Tensor[Float], val target: SerializableIndexedSeq[RoiLabel],
+  val isCrowd: SerializableIndexedSeq[Tensor[Float]], val imageInfo: Tensor[Float] = null)
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/transform/vision/image/MTImageFeatureToBatch.scala]: Array by default support serialization, right ?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/SerializableIndexedSeq.scala [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+package com.intel.analytics.bigdl.utils
+
+import java.io.{IOException, ObjectInputStream, ObjectOutputStream}
+import scala.reflect.ClassTag
+import scala.language.implicitConversions
+import scala.collection.mutable.{IndexedSeq, WrappedArray}
+
+
+class SerializableIndexedSeq[T: ClassTag](@transient private var impl: IndexedSeq[T])
```
Comment (zhangxiaoli73) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/SerializableIndexedSeq.scala]: private

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/transform/vision/image/MTImageFeatureToBatch.scala (hunk 1 above)
Comment (Menooker) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/transform/vision/image/MTImageFeatureToBatch.scala]: But the `slice` method of RoiMiniBatch uses array.slice, may not be efficient.
One of the old commit uses IndexSeq and array.view to achieve better performance but it causes serialization problems.

**Comment by Menooker** [comment]
Jenkins pass http://mlt-bddl01.sh.intel.com:8080/job/BigDL-PRVN-lite/1026/

**Comment by Menooker** [comment]
SerialiableIndexSeq removed, changed back to array.slice.
This commit now only resolves the problem of empty "target" and bug in RoiMiniBatch.slice

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/transform/vision/image/MTImageFeatureToBatchSpec.scala [hunk 3]
```diff
@@ -234,6 +234,30 @@ class MTImageFeatureToBatchSpec extends FlatSpec with Matchers with BeforeAndAft
     })
     imgCheck.count(!_) should be (0)
 
+  }
+  "MTImageFeatureToBatch without labels" should "work well" in {
+    val imgData = (0 to 1000).map(idx => (idx to (idx + 10*10*3)).map(_.toFloat).toArray)
+      .map(arr => {
+        val imf = ImageFeature()
+        imf(ImageFeature.floats) = arr
+        imf(RoiImageInfo.ISCROWD) = Tensor(Array(0f, 1f), Array(2))
```
Comment (zhangxiaoli73) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/transform/vision/image/MTImageFeatureToBatchSpec.scala]: no crowd

**Comment by Menooker** [comment]
Jenkins pass http://mlt-bddl01.sh.intel.com:8080/job/BigDL-PRVN-lite/1038/

**Pull request merged**